cf-execd/cf-execd-runner.c

```diff
@@ -337,21 +337,21 @@ void LocalExec(const ExecConfig *config)
 
     free(line);
     cf_pclose(pp);
-    Log(LOG_LEVEL_DEBUG, "Closing fp");
-    fclose(fp);
-
     Log(LOG_LEVEL_VERBOSE,
         complete ? "Command is complete" : "Terminated command");
 
     if (count)
     {
+        Log(LOG_LEVEL_DEBUG, "Closing fp");
+        fclose(fp);
```

Comment:
Why is this moved?

Reply:
Because I want to close the file _after_ using `unlink` in the `else` case.

Reply:
Because I wanted to close the file _after_ unlinking it in the else case.

Reply:
Then I'd move it **after** the whole `if-else` block.

Reply:
But the thing is, I need to close it before it is opened within `MailResult`. That was the reason `cf-execd` failed earlier.

A better solution might be to pass the `FILE`-pointer to MailResult.

Reply:
Makes sense.

> A better solution might be to pass the FILE-pointer to MailResult.

TODO.